lib/src/util.dart

```diff
@@ -423,6 +423,22 @@ Iterable<AstNode> allDescendants(AstNode node) sync* {
 Iterable<T> allDescendantsOfType<T extends AstNode>(AstNode node) =>
     allDescendants(node).whereType<T>();
 
+/// Finds a dependency within a pubspec and returns the [VersionRange].
+///
+/// NOTE: This logic is not very flexible and doesn't work for hosted dependencies.
+VersionRange? getDependencyVersion(String pubspecContent, String dependency) {
+  final dependencyValidatorRegex =
+      RegExp('^  $dependency: (.+)\$', multiLine: true);
```

Comment:
This whitespace handling could be a little more lenient
```suggestion
      RegExp('^ {2,}$dependency: *(.+)\$', multiLine: true);
```